firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistribution.java

```diff
@@ -126,8 +131,8 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
    * to complete the download and installation.
    */
   @NonNull
-  public synchronized UpdateTask updateIfNewReleaseAvailable() {
-    synchronized (updateTaskLock) {
+  public UpdateTask updateIfNewReleaseAvailable() {
```

Comment:
I spoke offline with @mannyjimenez0810 and decided to remove the synchronized methods in this class because:

- Using a synchronized block within a synchronized method increases the risk of deadlock
- I'm not sure how much benefit the synchronized method gives us in our use case. The developer can only create one task per-public method because we return the same task if the task is not complete.

Reply:
💯